ecl/eclccserver/eclccserver.cpp

```diff
@@ -303,19 +303,38 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
         query->getQueryText(eclQuery);
         query->getQueryMainDefinition(mainDefinition);
 
+        bool syntaxCheck = (workunit->getAction()==WUActionCheck);
```

Comment:
What happened before this fix if action was "syntaxCheck"? It would do a full compile?

Reply:
I think so.  That's what it appeared to be doing when I first tried.  I think more importantly it didn't support the option to specify the module/procedure.